packages/runtime/lib/start.js

```diff
@@ -70,8 +70,51 @@ async function startWithConfig (configManager) {
     async restart () {
       worker.postMessage({ msg: 'plt:restart' })
       await once(worker, 'message') // plt:restarted
+    },
+    async getServicesTopology () {
+      worker.postMessage({ msg: 'plt:get-topology' })
+      const [msg] = await once(worker, 'message')
+
+      const res = JSON.parse(msg.res)
+      if (res.error) {
+        throw new Error(res.error)
+      }
+
+      return res
+    },
+    async getServiceState (id) {
+      worker.postMessage({ msg: 'plt:get-state', params: { id } })
+      const [msg] = await once(worker, 'message')
+
+      const res = JSON.parse(msg.res)
+      if (res.error) {
+        throw new Error(res.error)
+      }
+
+      return res
+    },
+    async getServiceConfig (id) {
+      worker.postMessage({ msg: 'plt:get-config', params: { id } })
+      const [msg] = await once(worker, 'message')
+
+      const res = JSON.parse(msg.res)
+      if (res.error) {
+        throw new Error(res.error)
+      }
+
+      return res
+    },
+    async startService (id) {
+      worker.postMessage({ msg: 'plt:start-service', params: { id } })
+      await once(worker, 'message')
+    },
+    async stopService (id) {
+      worker.postMessage({ msg: 'plt:stop-service', params: { id } })
+      await once(worker, 'message')
     }
   }
```

Comment:
Can you add an API to call `inject()` on a given service?

Reply:
`inject()` does not really work across the thread boundary. We can only really support a very crippled version.

Reply:
Actually, we might be able to do this. I didn't realize that web streams were transferrable.